tests/test-cases/Group23-VIC-Machine-Service/23-06-VCH-Certificate.robot

```diff
@@ -0,0 +1,95 @@
+# Copyright 2017 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+*** Settings ***
+Documentation  Test 23-06 - VCH Certificate
+Resource  ../../resources/Util.robot
+Suite Setup  Start VIC Machine Server
+Suite Teardown  Terminate All Processes  kill=True
+Test Teardown  Cleanup VIC Appliance On Test Server
+Default Tags
+
+*** Keywords ***
+Install VIC Machine Without TLS
+    [Arguments]  ${vic-machine}=bin/vic-machine-linux  ${appliance-iso}=bin/appliance.iso  ${bootstrap-iso}=bin/bootstrap.iso  ${certs}=${true}  ${vol}=default  ${cleanup}=${true}  ${debug}=1  ${additional-args}=${EMPTY}
+    Set Test Environment Variables
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run  govc host.esxcli network firewall set -e false
+    # Attempt to cleanup old/canceled tests
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling VMs On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Datastore On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling Networks On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling vSwitches On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling Containers On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling Resource Pools On Test Server
+    Set Suite Variable  ${vicmachinetls}  --no-tls
+    Log To Console  \nInstalling VCH to test server with tls disabled...
+    ${output}=  Run VIC Machine Command  ${vic-machine}  ${appliance-iso}  ${bootstrap-iso}  ${certs}  ${vol}  ${debug}  ${additional-args}
+    Log  ${output}
+    Should Contain  ${output}  Installer completed successfully
+
+    Get Docker Params  ${output}  ${certs}
+    Log To Console  Installer completed successfully: %{VCH-NAME}...
+
+    [Return]  ${output}
+
+Start VIC Machine Server
+    Start Process  ./bin/vic-machine-server --port 31337  shell=True  cwd=/go/src/github.com/vmware/vic
+    Sleep  1s  for service to start
```

Comment:
can we do something better than this?  I see a race condition if the system is slow... is there a basic curl we can do that determines the server is up?   wrapped in a "Wait until keyword succeeds"

Reply:
How about we wrap the curl in a `Try n times until command succeeds`? A `/ping`-like endpoint does not exist in the service and adding one is not within the scope of this task. As such, any use of `curl` in this PR is essentially a retry curl against a testable, functional endpoint (like the one implemented here).

Reply:
I think the worry about a race condition is slightly overblown, as we would have to wait until the VCH is stood up before attempting to curl the service. If the system is slow, that would be a long wait. The service itself is pretty lightweight, small and fast.